sc/src/main/java/uk/ac/ebi/atlas/search/JsonGeneSearchController.java

```diff
@@ -123,26 +123,46 @@ private JsonObject facetValueObject(String group, String value) {
     }
 
     // Converts list of (k, clusterId) pairs into json objects
-    private JsonArray convertMarkerGeneModel(List<Pair<Integer, Integer>> model) {
+    private JsonArray convertMarkerGeneModel(String experimentAccession, String geneId, List<Pair<Integer, Integer>> model) {
         JsonArray result = new JsonArray();
 
         model.forEach((kClusterIdPair) ->
-                result.add(markerGeneObject(kClusterIdPair.getLeft(), kClusterIdPair.getRight()))
+                result.add(markerGeneObject(
+                        kClusterIdPair.getLeft(),
+                        kClusterIdPair.getRight(),
+                        createResultsPageURL(experimentAccession, geneId, kClusterIdPair.getLeft(), kClusterIdPair.getRight())))
         );
 
         return result;
     }
 
-    private JsonObject markerGeneObject(Integer k, Integer clusterId) {
+    private JsonObject markerGeneObject(Integer k, Integer clusterId, String url) {
         JsonObject result = new JsonObject();
 
         result.addProperty("k", k);
         result.addProperty("clusterId", clusterId);
+        result.addProperty("url", url);
 
         return result;
     }
 
-    private String createExperimentPageRelativePath(String contextPath, String experimentAccession) {
-        return String.join("/", contextPath, "experiments", experimentAccession);
+    private String createExperimentPageURL(String experimentAccession, String geneId) {
+       return ServletUriComponentsBuilder.fromCurrentContextPath()
```

Comment:
I was about to write in the commit with `String.join` that there must be libraries to manipulate URIs in a sane way. I was thinking in more general terms, but I like this solution. The fact that it has `fromCurrentContextPath` makes it all the better. 👍